src/main/java/com/diplomaProject/StudyBe/User/Service/UserServiceIpl.java

```diff
@@ -44,14 +47,26 @@ public User findById(Long id ) {
 
     @Override
     public User findByEmail(String email) {
-        List<User> userList = findAll();
-        Optional<User> userWithThatEmail = userList.stream()
-                .filter(user -> user.getEmail().equals(email))
-                .findFirst();
+//        List<User> userList = findAll();
+//        Optional<User> userWithThatEmail = userList.stream()
+//                .filter(user -> user.getEmail().equals(email))
+//                .findFirst();
 
-        return userWithThatEmail.orElse(null);
+//        return userWithThatEmail.orElse(null);
+        return userRepository.findByEmail(email).orElse(null);
     }
 
+    @Override
+    public void addSubject(Subject subject, User user) {
+        System.out.println(subject);
+        System.out.println(user);
+        if(user != null && subject != null)
```

Comment:
You should not validate in the service if the user is present all validations should not hit the service if its comming form request.
Have a look at interceptors its a way to handle if a user is authorized or present in the system
https://www.tutorialspoint.com/spring_boot/spring_boot_interceptor.htm